content/docs/context.md

```diff
@@ -4,220 +4,92 @@ title: Context
-  }
-}
-
-class MessageList extends React.Component {
-  render() {
-    const color = "purple";
-    const children = this.props.messages.map((message) =>
-      <Message text={message.text} color={color} />
-    );
-    return <div>{children}</div>;
-  }
-}
-```
-
-In this example, we manually thread through a `color` prop in order to style the `Button` and `Message` components appropriately. Using context, we can pass this through the tree automatically:
-
-```javascript{6,13-15,21,28-30,40-42}
-import PropTypes from 'prop-types';
-
-class Button extends React.Component {
-  render() {
-    return (
-      <button style={{background: this.context.color}}>
-        {this.props.children}
-      </button>
-    );
-  }
-}
-
-Button.contextTypes = {
-  color: PropTypes.string
-};
-
-class Message extends React.Component {
-  render() {
-    return (
-      <div>
-        {this.props.text} <Button>Delete</Button>
-      </div>
-    );
-  }
-}
-
-class MessageList extends React.Component {
-  getChildContext() {
-    return {color: "purple"};
-  }
-
-  render() {
-    const children = this.props.messages.map((message) =>
-      <Message text={message.text} />
-    );
-    return <div>{children}</div>;
-  }
-}
-
-MessageList.childContextTypes = {
-  color: PropTypes.string
-};
+```js
+const {Provider, Consumer} = React.createContext(defaultValue);
 ```
 
-By adding `childContextTypes` and `getChildContext` to `MessageList` (the context provider), React passes the information down automatically and any component in the subtree (in this case, `Button`) can access it by defining `contextTypes`.
+Creates a `{ Provider, Consumer }` pair.
 
-If `contextTypes` is not defined, then `context` will be an empty object.
+Optionally accepts a default value to be passed to Consumers without a Provider ancestor.
 
-## Parent-Child Coupling
+### `Provider`
 
-Context can also let you build an API where parents and children communicate. For example, one library that works this way is [React Router V4](https://reacttraining.com/react-router):
+```js
+<Provider value={/* some value */}>
+```
 
-```javascript
-import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
+A React component that allows Consumers to subscribe to context changes.
 
-const BasicExample = () => (
-  <Router>
-    <div>
-      <ul>
-        <li><Link to="/">Home</Link></li>
-        <li><Link to="/about">About</Link></li>
-        <li><Link to="/topics">Topics</Link></li>
-      </ul>
+Accepts a `value` prop to be passed to Consumers that are descendants of this Provider. One Provider can be connected to many Consumers. Providers can be nested to override values deeper within the tree.
 
-      <hr />
+### `Consumer`
 
-      <Route exact path="/" component={Home} />
-      <Route path="/about" component={About} />
-      <Route path="/topics" component={Topics} />
-    </div>
-  </Router>
-);
+```js
+<Consumer>
+ { value => { /* render something based on the context value */ } }
```

Comment:
People will try to write

```js
{ value => { <div>hi</div> } }
```

Maybe let's remove the second curlies and just leave

```js
<Consumer>
  {value => /* render something based on the context value */}
</Consumer>
```

Also nit: no need for space before `value`.